includes/acl/compression/stream/write_stats.h

```diff
@@ -141,41 +143,74 @@ namespace acl
 
 		void* context = allocate_context(allocator);
 		Transform_32* lossy_pose_transforms = allocate_type_array<Transform_32>(allocator, num_bones);
+		uint8_t* cache_flush_buffer = allocate_cache_flush_buffer(allocator);
 
 		{
-			flush_data_cache(allocator);
-
-			ScopeProfiler timer;
+			double total_elapsed = 0.0;
 
 			for (uint32_t sample_index = 0; sample_index < num_samples; ++sample_index)
 			{
 				float sample_time = min(float(sample_index) / sample_rate, clip_duration);
+
+				flush_cache(cache_flush_buffer);
+
+				ScopeProfiler timer;
 				decompress_pose(context, sample_time, lossy_pose_transforms, num_bones);
-			}
+				timer.stop();
 
-			timer.stop();
+				total_elapsed += timer.get_elapsed_seconds();
+			}
 
-			writer["forward_decompression_time"] = timer.get_elapsed_seconds();
+			writer["decompression_time_forward_playback"] = total_elapsed;
 		}
 
 		{
-			flush_data_cache(allocator);
-
-			ScopeProfiler timer;
+			double total_elapsed = 0.0;
 
 			for (int32_t sample_index = num_samples - 1; sample_index >= 0; --sample_index)
 			{
 				float sample_time = min(float(sample_index) / sample_rate, clip_duration);
+
+				flush_cache(cache_flush_buffer);
+
+				ScopeProfiler timer;
 				decompress_pose(context, sample_time, lossy_pose_transforms, num_bones);
+				timer.stop();
+
+				total_elapsed += timer.get_elapsed_seconds();
 			}
 
-			timer.stop();
+			writer["decompression_time_backward_playback"] = total_elapsed;
+		}
+
+		{
+			double total_elapsed = 0.0;
+
+			for (uint32_t seek_index = 0; seek_index < NUM_RANDOM_SEEKS; ++seek_index)
```

Comment:
We don't actually need to randomly seek since we destroy the context every time. We could simply iterate over every sample like we do above and delete/recreate the context everytime which would simulate the same behavior with a much more predictable pattern. Later, by keeping track of this data per sample time, we'll be able to track random seek performance everywhere in the clip.

Reply:
This is a really good idea.